src/gsts3multipartuploader.cpp

```diff
@@ -481,9 +482,14 @@ bool MultipartUploader::_init_uploader(const GstS3UploaderConfig * config)
     }
     client_config.verifySSL = config->aws_sdk_verify_ssl;
 
-    _s3_client = config->aws_sdk_s3_sign_payload ?
-        std::unique_ptr<Aws::S3::S3Client>(new Aws::S3::S3Client(std::move(credentials_provider), client_config)) :
-        std::unique_ptr<Aws::S3::S3Client>(new Aws::S3::S3Client(std::move(credentials_provider), client_config, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy::Never, false));
+    const char* endpoint_provider_allocation_tag = "AWSS3EndpointProvider";
+
+    if (config->aws_sdk_s3_sign_payload) {
+        client_config.payloadSigningPolicy = Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy::Never;
```

Comment:
I see that before we were setting payloadSigningPolicy to `Never` when `aws_sdk_s3_sign_payload` was `false`; now we do that when `aws_sdk_s3_sign_payload` is `true`. Could you clarify why is that?

Reply:
This is not intended, have changed to check for false now and ran some local executions to confirm behaviour. Will add a commit.